monitor_correct_errors/models/monitor_correct_errors.py

```diff
@@ -0,0 +1,336 @@
+        :return: -
+        '''
+
+        # Create the new hold
+        hold = self.env['compassion.hold'].create({
+            'expiration_date': self.env['compassion.hold'].
+                get_default_hold_expiration(HoldType.NO_MONEY_HOLD),
+            'child_id': child.id,
+            'type': HoldType.NO_MONEY_HOLD.value
+        })
+
+        child.hold_id = hold
+        try:
+            # Confirm within the global server if the child is available.
+            hold.update_hold()
+            child.hold_id = hold
+            # Commit when a hold is placed
+            self.env.cr.commit()  # pylint: ignore=invalid-commit
+
+            error_log['action'] = "new hold created"
+            error_log['action_model'] = "compassion.hold"
+            error_log['action_id'] = hold.id
+            error_log['action_name'] = "hold"
+
+        except:
+            # Create a reservation if hold has not been placed.
+            self.env.invalidate_all()
+            self._create_reservation(child, error_log)
+
+    def _create_reservation(self, child, error_log):
+        '''
+        Create a new reservation.
+        :param child: the child to reserve
+        :return: -
+        '''
+
+        # Check if there is already a reservation for that child.
+        previous_reservation = self.env['compassion.reservation'].search([
+            ('child_global_id', '=', child.global_id),
+            ('state', '!=', 'expired')])
+
+        # If not, we create a new reservation
+        if not previous_reservation:
+            reservation = self.env['compassion.reservation'].create({
+                'child_id': child.id,
+                'reservation_type': 'child',
+                'reservation_expiration_date': self.env[
+                    'compassion.hold'].get_default_hold_expiration(
+                    HoldType.CONSIGNMENT_HOLD),
+                'expiration_date': self.env[
+                    'compassion.hold'].get_default_hold_expiration(
+                    HoldType.CONSIGNMENT_HOLD),
+                'comments': "Automatic reservation place for missing hold"
+            })
+            # Commit when we create a reservation to avoid its deletion
+            self.env.cr.commit()    # pylint: ignore=invalid-commit
+            # Activation of the reservation
+            reservation.handle_reservation()
+
+            error_log['action'] = "new reservation created"
+            error_log['action_model'] = "compassion.reservation"
+            error_log['action_id'] = reservation.id
+            error_log['action_name'] = "reservation"
+
+        else:
+            error_log['action'] = "reservation already exist"
+            error_log['action_model'] = "compassion.reservation"
+            error_log['action_id'] = previous_reservation.id
+            error_log['action_name'] = "reservation"
+
+    def _sent_mail(self):
+
+        message_body = ""
+        error_type_list = {}
```

Comment:
If you have a fields.Selection for the error types, you will be able to get the values with the call of a function you define on model error.log